src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/ILToLLVMImporter.cs

```diff
@@ -4035,7 +4037,15 @@ private void ImportConvert(WellKnownType wellKnownType, bool checkOverflow, bool
             }
             else
             {
-                LLVMValueRef converted = CastIfNecessary(loadedValue, GetLLVMTypeForTypeDesc(destType), value.Name(), wellKnownType == WellKnownType.UInt64 /* unsigned is always false, so check for the type explicitly */);
+                TypeDesc sourceType = value.Type;
+
+                LLVMValueRef converted = CastIfNecessary(loadedValue, GetLLVMTypeForTypeDesc(destType), value.Name(),
+                    wellKnownType == WellKnownType.UInt64 /* unsigned is always false, so check for the type explicitly */
+                    || sourceType.IsWellKnownType(WellKnownType.Byte)
+                    || sourceType.IsWellKnownType(WellKnownType.UInt16)
+                    || sourceType.IsWellKnownType(WellKnownType.UInt32)
+                    || sourceType.IsWellKnownType(WellKnownType.UInt64)
+                    );
```

Comment:
```suggestion
                LLVMValueRef converted = CastIfNecessary(loadedValue, GetLLVMTypeForTypeDesc(destType), value.Name(), unsigned);
```
Presumably that should now work?

I also see there is:
```
            else if ((toStoreKind == LLVMTypeKind.LLVMDoubleTypeKind || toStoreKind == LLVMTypeKind.LLVMFloatTypeKind) &&
                valueTypeKind == LLVMTypeKind.LLVMIntegerTypeKind)
            {
                //TODO: keep track of the TypeDesc so we can call BuildFPToUI when the integer is unsigned
                typedToStore = builder.BuildFPToSI(source, valueType, "CastFloatSI" + (name ?? ""));
            }
```
In `CastIfNecessary` that should presumably also be fixed to look at `unsigned` for the new test to pass.